applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,395 @@
+# FSA technical specification
+
+This document contains the technical specifications for Feedback Sentiment Analyzer (FSA), a sample application that showcases AWS services and SDKs.
+
+This document explains the following:
+
+- Application inputs and outputs
+- Underlying AWS components and their configurations
+- Key cross-service integration details
+
+For an introduction to FSA, see the [README.md](README.md).
+
+---
+
+### Table of contents
+
+- [Diagram](#diagram)
+- [User actions](#user-actions)
+- [Application behavior](#application-behavior)
+- [HTTP API specification](#http-api-specification)
+- [Step Function configuration](#step-function-configuration)
+  - [State machine Lambda functions](#state-machine-lambda-functions)
+    - [ExtractText](#extracttext)
+    - [AnalyzeSentiment](#analyzesentiment)
+    - [TranslateText](#translatetext)
+    - [SynthesizeAudio](#synthesizeaudio)
+- [GetFeedback Lambda function](#getfeedback-lambda-function)
+- [Processing Amazon S3 events with EventBridge](#processing-s3-events-with-eventbridge)
+- [Managing items in DynamoDB](#managing-items-in-dynamodb)
+- [Other FSA material](#other-fsa-material)
+
+---
+
+## Relational diagram
+
+This diagram represents the relationships between key FSA components.
+![relational diagram](docs/architecture2.png)
+
+1. The static website assets are hosted in an Amazon S3 bucket and served using Amazon CloudFront.
+2. Amazon Cognito allows authenticated access to Amazon API Gateway.
+3. Amazon API Gateway puts objects in an Amazon S3 bucket. This triggers an EventBridge rule that starts a Step Functions workflow.
+4. The Step Functions workflow uses AWS Lambda, Amazon Textract, Amazon Comprehend, Amazon Translate, and Amazon Polly to perform the business logic.
+5. Metadata is stored in Amazon DynamoDB. Audio files are stored in the same Amazon S3 bucket used in step 3.
+6. Amazon API Gateway fetches the metadata from Amazon DynamoDB.
+
+---
+
+## User actions
+
+This application receives three inputs from the frontend.
+
+- Authenticate user (uses `Cognito`)
+- Load page (uses `DynamoDB:GetItems`)
+- Upload image to Amazon S3 (`S3:PutObject`)
+
+## Application behavior
+
+This application produces two outputs from the backend:
+
+- Put new item in the database (uses `DynamoDB:PutItem`)
+- Put new synthesized audio to Amazon S3 (uses `S3:PutObject`)
+
+---
+
+## HTTP API specification
+
+All the APIs are created by the CDK script. The endpoints are common to every language variation and do not need any additional implementation.
+
+<details>
+<summary><strong>PUT /api/media/{item}</strong></summary>
+<br/>
+
+> Create or update an object in Amazon S3. Creating or updating an image file will trigger the Step Functions workflow.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>item - the object key of the item to get</td>
+<td>JavaScript [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object</td>
```

Comment:
For some reason, the `File` link is rendering as the raw text instead of the link text with underlying URL.